### ISSUE TITLE
Block/Tx: Interoperability with Ethers v6

### ISSUE THREAD

**Issue by holgerd77** [ethereumjs/ethereumjs-monorepo#1099]
Depends on #1048 

Another Ethers issue 😄 : I lately had the use case where I wanted to pull in some tx data via RPC with Ethers which I wanted to use in some client testing context but I finally gave up since the conversion to [our tx format here](https://github.com/ethereumjs/ethereumjs-monorepo/tree/master/packages/tx) was not so clearly spottable and I used some direct client data extraction technique. Not that it was not possible, but it was just not obvious how to convert (likely calling some serialization method on the [Ethers transaction](https://docs.ethers.io/v5/api/utils/transactions/) and then reading in with the respective static constructor method on our side).

I think it would be convenient to have a direct `Tx.fromEthers()` constructor doing these steps internally to improve on the interoperability of the tx library formats. Not sure if it would make sense to also have this the other way around, in most cases I assume it's likely some use case similar to the one described above? But not sure.

//cc @ricmoo circling you in here, do you think we could do a generic method like above or would you rather recommend to make this version specific, so e.g. `Tx.fromEthersV5()`?

This will generally likely also be influenced by the typed transaction implementation from #1048 - not sure how far an implementation is on the Ethers side here - and we should likely also tackle this issue after merging the above PR.

**Comment by holgerd77** [comment]
(note that this generally just some nice-to-have thing absolutely not being urgent, just writing it down since I just recalled on the situation and head this in my head 😄 )

**Comment by ricmoo** [comment]
I'm working on the v6 Transaction class right now. What parts are incompatible? I'm guessing the name `gasLimit` (ethers) vs `gas`? Is there anything else?

**Comment by holgerd77** [comment]
I think we won't get interoperability on the object level with parameters matching respectively an EthereumJS tx directly used in Ethers and vice versa (or should we try?), also the surrounding functionality and API is likely implemented too differently.

I was just thinking moderately about wrapping the serialization call on the Ethers side and the call our respective `fromSerializedTx()` function subsequently for user convenience. Just looking though, might not be as easy as one might think, on the Ethers side you can't do anything like `tx.serialize()` but use [ethers.utils.serializeTransaction()](https://docs.ethers.io/v5/api/utils/transactions/#utils-serializeTransaction), right? 🤔

That likely wouldn't make up on such a good API on our side with the Ethers object itself also need to be passed in?

**Comment by holgerd77** [comment]
As a super low-level solution we could just add such a use case to our documentation.

**Comment by ricmoo** [comment]
Oh, true. You could just use the `serializeTransaction` function. In v6, Transaction is a bona-fide class, which does have a `.serialized` getter (looks like a normal property ;)).

**Comment by holgerd77** [comment]
Still think it would be nice to improve on this 🙂, have expanded the issue to include block compatibility as well.

**Comment by acolytec3** [comment]
We've got probably 90% of what is needed to convert blocks in our helpers in ultralight now and maybe 75% of what you need for transactions too. https://github.com/ethereumjs/ultralight/blob/master/packages/portalnetwork/src/util/helpers.ts

**Comment by holgerd77** [comment]
Yeah, I would also think this should not be such a big deal and somewhat like 1-2 days work (at most).

There are rather a few strategic questions to settle in a first round:

1. Generic name functions `fromEthers()` or version-based ones `fromEthersV6()`?
2. Where to put Ethers as a dependency (either in Block/Tx or - also possible - in Util as a re-export) (though we moved a bit away from this with the latest RLP, ethereum-cryptography *extraction*). It generally might be a good idea to just have a certain-version-Ethers-version available to be integrated monorepo-wide (more or less) to be used in the various libraries, it seems we are using Ethers in more and more places for targeted things, would assume we'll find even more use cases for this in the future

**Comment by holgerd77** [comment]
> We've got probably 90% of what is needed to convert blocks in our helpers in ultralight now and maybe 75% of what you need for transactions too. https://github.com/ethereumjs/ultralight/blob/master/packages/portalnetwork/src/util/helpers.ts

Ah, and generally: that's great to hear, then it would make even more sense to extract this functionality and make this generic and use the specific internalized functions.

**Comment by acolytec3** [comment]
Do we still need this? We've added and then removed Ethers as a dependency in `block` and `tx`. If we do need something like this, we have something like this already implemented in ultralight we could adapt.

**Comment by holgerd77** [comment]
Will close for now, seemed not too pressing to implement. If someone feels like it just do (without adding dependencies).
